File: src/SignalR/common/Protocols.Json/src/JsonHubProtocolOptions.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

using System.Text.Json;
using Microsoft.AspNetCore.SignalR.Protocol;

namespace Microsoft.AspNetCore.SignalR;

/// <summary>
/// Options used to configure a <see cref="JsonHubProtocol"/> instance.
/// </summary>
public class JsonHubProtocolOptions
{
    /// <summary>
    /// Gets or sets the settings used to serialize invocation arguments and return values.
    /// </summary>
    public JsonSerializerOptions PayloadSerializerOptions { get; set; } = JsonHubProtocol.CreateDefaultSerializerSettings();
}

File: src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/JsonHubProtocolOptions.java
package com.microsoft.signalr;

import com.google.gson.GsonBuilder;

/**
 * Used to configure options for the creation of the JSON hub protocol
 */
public class JsonHubProtocolOptions {

    private GsonBuilder gsonBuilder;

    /**
     * Creates a default instance with no changes to the default configuration
     */
    public JsonHubProtocolOptions() {
    }

    /**
     * Creates an instance with the specified {@link GsonBuilder} which be used to deserialise
     * user payloads.
     */
    public JsonHubProtocolOptions(GsonBuilder gsonBuilder) {
        this.setGsonBuilder(gsonBuilder);
    }

    /**
     * Gets the {@link GsonBuilder} which be used to deserialise user payloads, if provided.
     */
    public GsonBuilder getGsonBuilder() {
        return gsonBuilder;
    }

    /**
     * Sets the {@link GsonBuilder} which be used to deserialise user payloads, or null if you
     * want to use the default {@link GsonBuilder}.
     */
    public void setGsonBuilder(GsonBuilder gsonBuilder) {
        this.gsonBuilder = gsonBuilder;
    }
}
